CHANGES.md

```diff
@@ -1,6 +1,429 @@
+* added support for printing 'c_ptr' and 'c_void_ptr' variables
+* added support for an 'isExternClassType()` query
+
+Performance Optimizations/Improvements
+--------------------------------------
+* registered arrays with the network dynamically for better NUMA locality
+* significantly improved the ability of the LLVM back-end to optimize Chapel
+* improved locality for sparse Block-distributed domains/arrays
+* optimized bulkAdd calls for empty sparse domains
+* reduced the amount of locking used for associative array accesses
+* added support for serializing records across locales, supporting local copies
+* optimized remote task creation for cases like 'coforall ... do on ... {}'
+* automatically inline iterators that have up to 'k' yields (10 by default)
+* improved the performance of dynamic casting / subclass checks
+* improved wide-pointer analysis for 'const ref' arguments
+* optimized away some wide pointer overheads in 'Block' and 'Stencil'
+* fixed bugs in counting tasks and creating the right number of new tasks
+* added support for parallel array initialization for arrays of POD eltTypes
+* optimized the implementation of updateFluff() on the Stencil distribution
+* privatized domains and distributions created by rank changes and reindexing
+* made the domains and distributions of reindexed arrays preserve locality
+
+Memory Improvements
+-------------------
+* stopped heap-promoting local variables used within on-clauses for 'qthreads'
+* closed memory leaks in the 'Regexp' module and DimensionalDist2D distribution
+
+Compiler Flags
+--------------
+(see 'man chpl' or http://chapel.cray.com/docs/1.16/usingchapel/man.html for details)
+* added --permit-unhandled-module-errors to enable relaxed error checking
+* added --print-unused-functions to identify unused routines
+* added --[no-]remote-serialization to control cross-locale record copies
+* added an --mllvm flag to set LLVM-based optimizations when using --llvm
+* added --inline-iterators-yield-limit to cap # of yields to inline for iters
+* gave the --cache-remote flag a --no-cache-remote variation
+* removed the --strict-errors flag in favor of new more precise features
+* removed the --conditional-dynamic-dispatch-limit flag and feature
+
+Documentation
+-------------
+* added new primers on error-handling and the 'LinearAlgebra' module
+  (see http://chapel.cray.com/docs/1.16/primers/errorHandling.html and
+   http://chapel.cray.com/docs/1.16/primers/LinearAlgebralib.html)
+* added a document describing built-in `Error` types in Chapel
+  (see http://chapel.cray.com/docs/1.16/builtins/internal/ChapelError.html)
+* broke information about the `Replicated` distribution into its own primer
+  (see http://chapel.cray.com/docs/1.16/primers/replicated.html)
+* added a new 'Methods' chapter to the language spec and refreshed the content
+* improved the language specification's definition of records
+* documented the ability to specify configuration files via '-f'
```

Comment:
Searching on "comment" and "config", I didn't see a mention of the new comment support for these configuration files (#6583)

Reply:
Good catch.  Note that the best way to ensure changes get mentioned in the CHANGES file is to advertise them in the merge message's headline.  If I read it and understand it, I don't read the rest of the message.

Reply:
Oh wait, I take that back—my bad.  I mis-read the headline (both times) as being about the documentation of the support.  Maybe because Paul mentioned that just the other week...